go/cli/internal/cli/init.go

```diff
@@ -0,0 +1,97 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/kagent-dev/kagent/go/api/v1alpha2"
+	"github.com/kagent-dev/kagent/go/cli/internal/config"
+	"github.com/kagent-dev/kagent/go/cli/internal/frameworks"
+)
+
+type InitCfg struct {
+	Framework       string
+	Language        string
+	AgentName       string
+	InstructionFile string
+	ModelProvider   string
+	ModelName       string
+	Description     string
+	Config          *config.Config
+}
+
+func InitCmd(cfg *InitCfg) error {
+	// Validate framework and language
+	if cfg.Framework != "adk" {
+		return fmt.Errorf("unsupported framework: %s. Only 'adk' is supported", cfg.Framework)
+	}
+
+	if cfg.Framework == "adk" && cfg.Language != "python" {
+		return fmt.Errorf("unsupported language: %s. Only 'python' is supported for ADK", cfg.Language)
+	}
+
+	if cfg.ModelName != "" && cfg.ModelProvider == "" {
+		return fmt.Errorf("model provider is required when model name is provided")
+	}
+
+	// Validate model provider if specified
+	if cfg.ModelProvider != "" {
+		if err := validateModelProvider(cfg.ModelProvider); err != nil {
+			return err
+		}
+	}
+
+	if cfg.ModelProvider == "" && cfg.ModelName == "" {
+		cfg.ModelProvider = string(v1alpha2.ModelProviderGemini)
+		cfg.ModelName = "gemini-2.0-flash"
+	}
```

Comment:
I don't think we should default here, it's a little too user specific.

Reply:
Removed the default here but added defaults for the `--model-name` and `--model-provider` flags instead.